openedx/core/djangoapps/user_api/views.py

```diff
@@ -1143,3 +1162,23 @@ class CountryTimeZoneListView(generics.ListAPIView):
     def get_queryset(self):
         country_code = self.request.GET.get('country_code', None)
         return get_country_time_zones(country_code)
+
+
+class HintUsernameView(APIView):
+    """
+    DRF APIView to get a username suggestion.
+    If the user exists then suggests a new username, otherwise
+    returns the received base username.
+    """
+
+    def get(self, request):
+        """
+        Returns a valid username suggestion checking against the database.
+        """
+        data_serializer = HintUsernameSerializer(data=request.query_params)
+        data_serializer.is_valid(raise_exception=True)
+        user_dict = data_serializer.data
+        username = user_dict['username']
+
+        new_username = UsernameGenerator().generate_username(username)
+        return Response({'username': new_username})
```

Comment:
Since you've moved the username hint logic into https://github.com/edx-olive-oc/campus-social-auth/pull/1 (thank you for that!), you'll need to move this endpoint there too.  Here's how I suggest to do it:
* move the [HintUsernameView](https://github.com/open-craft/edx-platform/pull/120/files#diff-224fd193a6c7eba663b3764afec7110dR1167) and [HintUsernameSerializer](https://github.com/open-craft/edx-platform/pull/120/files#diff-cbde7a82f34bc0f643dddeb27ba31062R113) into `campus-social-auth`, something [like this](https://github.com/edx-olive-oc/campus-social-auth/commit/f7d6a803446432c40fd76e96dbcaeaad54aa0a68)
* Add the URL to the LMS conditionally, like this:
```diff
diff --git a/lms/urls.py b/lms/urls.py
index 3fb0ea373c..4967e29fe1 100644
--- a/lms/urls.py
+++ b/lms/urls.py
@@ -926,6 +926,14 @@ if settings.FEATURES.get('ENABLE_THIRD_PARTY_AUTH'):
         url(r'^login_oauth_token/(?P<backend>[^/]+)/$', 'student.views.login_oauth_token'),
     )
 
+    for backend in settings.AUTHENTICATION_BACKENDS:
+        if 'campus_social_auth' in backend:
+            urlpatterns += (
+                url(r'^campus_social_auth/', include('campus_social_auth.urls')),
+            )
+            break
+
+
 # Enterprise
 if enterprise_enabled():
     urlpatterns += (
```

This will be some code drift for now, but once we upgrade to Hawthorn, we can make `campus-social-auth` be a [django plugin app](https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/plugins/README.rst), and then it can register its own URLs.